src/message-builder.js

```diff
@@ -1,53 +1,112 @@
-const slackifyMarkdown = require('slackify-markdown');
+const slackifyMarkdown = require("slackify-markdown");
 
 const buildMessage = (channel, text, time) => {
-    return { channel: channel, text: slackifyMarkdown(text), post_at: Date.parse(time) / 1000 }
-}
+  return {
+    channel: channel,
+    text: slackifyMarkdown(text),
+    post_at: Date.parse(time) / 1000,
+  };
+};
 
 function checkField(message, key) {
-    if (!message[key]) {
-        throw `${key} is missing in a message in ${message.file}`
+  if (!message[key]) {
+    throw `${key} is missing in a message in ${message.file}`;
+  }
+}
+function stringToDateConverter(messages) {
+  for (let message of messages) {
+    message.post_at = new Date(message.post_at);
+    if (message.repeat) {
+      message.repeat = new Date(message.repeat);
     }
+  }
+  return messages;
 }
 
 function convertChannelNameToId(channel, channels) {
-    for (let c of channels) {
-        if (c.id == channel) {
-            return channel;
-        }
-        if (c.name == channel) {
-            return c.id;
-        }
+  for (let c of channels) {
+    if (c.id == channel) {
+      return channel;
+    }
+    if (c.name == channel) {
+      return c.id;
     }
-    return null;
+  }
+  return null;
 }
 
+function incrimentDateIterative(startDate, endDate) {
+  let date = new Date(startDate);
+  let dateList = [];
+
+  while (date < endDate) {
+    dateList.push(date);
+    date = new Date(startDate.setDate(startDate.getDate() + 7));
```

Comment:
the startDate.setDate is not needed anymore

Reply:
do you mean the setDate() method? 
I am not sure how/why this happens, but without` startDate.setDate()` , and instead only  `Date(startDate.getDate() + 7)` , the date gets set back to `December 1969. ` 
I dont understand the behavior,  as the Date constructor begins on `Jan 1970...` . 
 And then for the rest of the program, it enters an infinite loop with the `date = Dec 31 1969...`

Reply:
Wouldn't 
```
  while (date < endDate) {
    dateList.push(date);
    date.setDate(date.getDate() + 7);
  }
  ```
  
  Work then?

Reply:
If I recall correctly, a new Date object has to be declared each time date is assigned with a new date. If not, while the `date` object is added to the array, and since it is one/the same object, with each iteration, all list items will be updated with that date. So in the end it would be an array of all the same dates.  I had some trouble understanding why the date object behaves the way it does, these are some of the things I've discovered..